apis/nucleus/src/hooks/__tests__/use-object-selections.inspect.jsx

```diff
@@ -0,0 +1,261 @@
+/* eslint no-import-assign: 0 */
+import React, { forwardRef, useImperativeHandle } from 'react';
+import { create, act } from 'react-test-renderer';
+
+import useObjectSelections from '../useObjectSelections';
+import * as useAppSelectionsModule from '../useAppSelections';
+import * as useLayoutModule from '../useLayout';
+import * as selectionStoreModule from '../../stores/selections-store';
+
+const TestHook = forwardRef(({ hook, hookProps = [] }, ref) => {
+  const result = hook(...hookProps);
+  useImperativeHandle(ref, () => ({
+    result,
+  }));
+  return null;
+});
+
+describe('useObjectSelections', () => {
+  let renderer;
+  let render;
+  let ref;
+  let modalObjectStore;
+  let app;
+  let appSel;
+  let appModal;
+  let object;
+  let objectSel;
+  let layout;
+
+  beforeEach(() => {
+    app = {
+      id: 'appSel',
+      session: {},
+      abortModal: jest.fn(),
+      forward: jest.fn(),
+      back: jest.fn(),
+      clearAll: jest.fn(),
+      getField: jest.fn(),
+    };
+    appModal = {
+      begin: jest.fn(),
+      end: jest.fn(),
+      _isMock: true,
+    };
+    appSel = {
+      isModal: jest.fn(),
+    };
+    layout = {};
+    modalObjectStore = {
+      set: jest.fn(),
+      get: jest.fn(),
+    };
+    object = {
+      beginSelections: jest.fn(),
+      endSelections: jest.fn(),
+      resetMadeSelections: jest.fn(),
+      clearSelections: jest.fn(),
+      select: jest.fn(),
+    };
+
+    jest.spyOn(useAppSelectionsModule, 'default').mockImplementation(() => [appSel]);
+    jest.spyOn(useLayoutModule, 'default').mockImplementation(() => [layout]);
+    jest.spyOn(selectionStoreModule, 'useAppSelectionsStore').mockImplementation(() => [
+      {
+        get: () => appSel,
+        set: (k, v) => {
+          appSel = v;
+        },
+      },
+    ]);
+    jest.spyOn(selectionStoreModule, 'useObjectSelectionsStore').mockImplementation(() => [
+      {
+        get: () => objectSel,
+        set: (k, v) => {
+          objectSel = v;
+        },
+        dispatch: async (b) => {
+          if (!b) return;
+          await act(async () => {
+            renderer.update(<TestHook ref={ref} hook={useObjectSelections} hookProps={[app, object]} />);
+          });
+        },
+      },
+    ]);
+    jest.spyOn(selectionStoreModule, 'useAppModalStore').mockImplementation(() => [{ get: () => appModal }]);
+    selectionStoreModule.modalObjectStore = modalObjectStore;
+
+    ref = React.createRef();
+    render = async () => {
+      await act(async () => {
+        renderer = create(<TestHook ref={ref} hook={useObjectSelections} hookProps={[app, object]} />);
+      });
+      objectSel.setLayout(layout);
+    };
+  });
+
+  afterEach(() => {
+    renderer.unmount();
+    jest.restoreAllMocks();
+    jest.resetAllMocks();
+  });
+
+  test('should create object selections', async () => {
+    await render();
+    expect(ref.current.result[0]).toEqual(objectSel);
+  });
+
+  test('should begin', async () => {
```

Comment:
This test file will entirely fail except 3 or 4 test cases!

The reason is, (my guess is) tests are not isolated and some tests are just bleeding to another and effect the run time of other tests. If you run tests individually -> they will pass but if you run them together they will not. 

I have spend like 2-3 hours just to figuring out what is causing this but no luck. I mock required things like previous implementation with aw, and [even in the actual file at here](https://github.com/qlik-oss/nebula.js/blob/495cf289c5fbd3748827e17072a41d58331f4a57/apis/nucleus/src/hooks/useObjectSelections.js#L46) if i log the `appModal.begin()` method, it will give me the **mocked function** and **clearly indicated that it is a mock function** while reaching to that line in my test environment, but for some unknown reason seems like it fails to pass below line: 
```js
expect(appModal.begin).toHaveBeenCalledTimes(1);
expect(appModal.begin).toHaveBeenCalledWith(object, ['/foo'], true);
```

I really have no idea why this happens and causes other tests to fail as well 🤷‍♂️ ☠️. for that reason i still keep the aw test file for this module.

Reply:
Ok, remind me later (when the rest of the conversion is done) to have a look at this

Reply:
I'll skip the failing test suit for now. Here is the reference for [original aw version of that test file](https://github.com/qlik-oss/nebula.js/blob/b5ba2d078e72459271aa04d3de945ed6db4a37c3/apis/nucleus/src/hooks/__tests__/use-object-selections.spec.jsx) in case if you want to fix it and looking for reference.